packages/orm/src/EntityManager.ts

```diff
@@ -398,6 +401,90 @@ export class EntityManager<C = {}> {
     return createOrUpdatePartial(this, type, opts);
   }
 
+  /**
+   * Utility to clone an entity and its nested relations, as determined by a populate hint
+   *
+   * @param entity - Any entity
+   * @param hint - A populate hint
+   *
+   * @example
+   * // This will duplicate the author
+   * const duplicatedAuthor = await em.clone(author)
+   *
+   * @example
+   * // This will duplicate the author and all their related book entities
+   * const duplicatedAuthorAndBooks = await em.clone(author, "books")
+   *
+   * @exmaple
+   * // This will duplicate the author, all their books, and the images for those books
+   * const duplicatedAuthorAndBooksAndImages = await em.clone(author, {books: "image"})
+   */
+  public async clone<T extends Entity, H extends LoadHint<T>>(entity: T, hint?: H): Promise<Loaded<T, H>> {
+    const meta = getMetadata(entity);
+    const { id, ...data } = entity.__orm.data;
+    const clone = this.create(meta.cstr, {} as OptsOf<T>);
+    clone.__orm.data = data;
+    if (hint) {
+      if ((typeof hint as any) === "string") {
+        hint = { [hint as keyof RelationsIn<T>]: {} } as any;
+      } else if (Array.isArray(hint)) {
+        hint = Object.fromEntries(hint.map((relation) => [relation, {}]));
+      }
+      await Promise.all(
+        (Object.entries(hint as NestedLoadHint<T>) as [keyof RelationsIn<T>, LoadHint<any>][]).map(
+          async ([relationName, nested]) => {
+            const relation = (entity[relationName] as any) as
+              | OneToManyCollection<T, any>
+              | OneToOneReference<T, any>
+              | ManyToOneReference<T, any, undefined | never>;
+            if (relation instanceof OneToManyCollection) {
+              const relatedEntities = await relation.load();
+              await Promise.all(
+                relatedEntities.map(async (related) => {
+                  const clonedRelated = await this.clone(related, nested);
+                  // Many to One relationships try and fetch their data from the cache if they can,
+                  // but the data is pointing to the uncloned entity, resulting in the original relation
+                  // being unset.
+                  // So we clear this data from the clone to ensure that the relation is set correctly
```

Comment:
Could shorten this comment to:

```
// Clear to avoid `set` mutating the original/source entity's relationship
```
 
especially since this comment block is copy/pasted ~3 times (which is otherwise fine)